pkg/daemons/control/deps/deps.go

```diff
@@ -294,7 +294,7 @@ func genClientCerts(config *config.Control, runtime *config.ControlRuntime) erro
 		return err
 	}
 
-	certGen, err = factory("cloud-controller-manager", nil, runtime.ClientCloudControllerCert, runtime.ClientCloudControllerKey)
+	certGen, err = factory(version.Program+"-cloud-controller-manager", nil, runtime.ClientCloudControllerCert, runtime.ClientCloudControllerKey)
```

Comment:
that will create a conflict with rke2 since in the manifest is statically set to k3s- not version.program, for reference check how we put a template variable to the manifests https://github.com/k3s-io/k3s/blob/master/pkg/server/server.go#L230 https://github.com/k3s-io/k3s/blob/master/manifests/coredns.yaml

Reply:
@galal-hussein yes, it is my expectation that RKE2 will need to be updated as well once this is pulled through. RKE2 does not use the ccm.yaml manifest that's embedded in k3s, instead the roles are created via bootstrap code at https://github.com/rancher/rke2/blob/master/pkg/rke2/clusterrole_templates.go#L6